cli/src/movement_util.rs

```diff
@@ -118,7 +146,24 @@ fn get_target_commit(
     working_commit_id: &CommitId,
     args: &MovementArgsInternal,
 ) -> Result<Commit, CommandError> {
-    let target_revset = direction.get_target_revset(working_commit_id, args)?;
+    let wc_revset = RevsetExpression::commit(working_commit_id.clone());
+    // If we're editing, start at the working-copy commit. Otherwise, start from
+    // its direct parent(s).
+    let start_revset = if args.should_edit {
+        wc_revset.clone()
+    } else {
+        wc_revset.parents()
+    };
+
+    let start_commits: Vec<Commit> = start_revset
+        .clone()
+        .evaluate_programmatic(workspace_command.repo().as_ref())?
+        .iter()
+        .commits(workspace_command.repo().store())
+        .try_collect()?;
+
+    let target_revset = direction.get_target_revset(&wc_revset, &start_revset, args)?;
```

Comment:
nit: wc commit can be looked up without using revset.

```rust
    let wc_commit = repo.store().get_commit(&working_commit_id)?;
    wc_commit.parents()?;
```

and the resulting commit ids can be passed in to `get_target_revset()` in place of `start_revset` and `wc_revset`.

Reply:
Hmmm... `get_target_revset` needs the `start_revset` and `wc_revset` so we can run expressions that return conflicted or non-conflicted ancestors/descendants.

The API suggest we can't do that with raw `Commit`.

Or did I completely misunderstand your suggestion?

Reply:
So, I'm still not getting it, I'm probably just slow. That said, I've gone ahead to rename `get_target_revset` -> `build_target_revset` and I've moved `start_commits` to be built just before we need it. I think this communicates the intents clearer. Maybe that helps us converge?

Based on that, we're not resolving `wc_revset` in `get_target_revset/build_target_revset`, we just use it when filtering children of the start_commits.

Also, I guess I could indeed resolve `start_commits`, then pass in the commits to `build_target_revset`, but then I would immediately construct a revset expression again. Though, I really don't need `start_commits` to be resolved anywhere else except for the `target_not_found_error` error message builder, so I've moved it there just before it's called.

So to check my understanding of the suggestion:
- `wc_revset` is never actually resolved, so I'm thinking no change there?
- `start_commit` is only resolved for the error message. Do you still think I should resolve, pass in the commits to `build_target_revset` and reconstruct the `start_revset` expression in there?

Reply:
> I've moved `start_commits` to be built just before we need it. I think this communicates the intents clearer. Maybe that helps us converge?
> 
> Based on that, we're not resolving `wc_revset` in `get_target_revset/build_target_revset`, we just use it when filtering children of the start_commits.

Yep, new code looks better.

BTW, we might want to make `target_not_found_error()` and `build_target_revset()` look similar because these two functions tests the same conditions `(Direction, should_edit, conflict)` to build revset to be evaluated and error message of the evaluation respectively, but I don't have a good idea.

Reply:
Great observation!

I've gone ahead to change `build_target_revset()` to look similar to `target_not_found_error()`, but there's a slight difference, since we use `should_edit` to get the `start_revset` outside `build_target_revset()`, before passing only `start_revset` in, so we no longer need `should_edit` in `build_target_revset()`.

I considered moving `should_edit` inside, only passing `wc_revset` alone and then having different revset chains, but it felt a bit too forced, especially since the chains look exactly the same, so I didn't go that route.

This definitely looks better though.

Thanks